app/models/campaign_page.rb

```diff
@@ -27,4 +30,9 @@ class CampaignPage < ActiveRecord::Base
   def create_slug
     self.slug = title.parameterize if slug.nil? and not title.nil?
   end
+
+  # Compiles the HTML for this CampaignPage so that it can be used by external display apps.
+  def compile_html
+    self.compiled_html = render partial: 'campaign_pages/page_compile', layout: false, locals: {campaign_page: self}
```

Comment:
Unsure about the side effects here - possibly I should just return the value instead?